src/main/java/seedu/address/model/Model.java

```diff
@@ -56,64 +56,94 @@ public interface Model {
     /** Returns the AddressBook */
     ReadOnlyAddressBook getAddressBook();
 
+
+    //// Person-related methods ////
+
     /**
      * Returns true if a person with the same identity as {@code person} exists in the address book.
      */
     boolean hasPerson(Person person);
 
-    boolean hasTutorial(Tutorial tutorial);
-
     /**
      * Deletes the given person.
      * The person must exist in the address book.
      */
     void deletePerson(Person target);
 
-
     /**
      * Adds the given person.
      * {@code person} must not already exist in the address book.
      */
     void addPerson(Person person);
 
-    void addTutorial(Tutorial tutorial);
-
     /**
      * Replaces the given person {@code target} with {@code editedPerson}.
      * {@code target} must exist in the address book.
      * The person identity of {@code editedPerson} must not be the same as another existing person in the address book.
      */
     void setPerson(Person target, Person editedPerson);
 
-    boolean hasParticipation(Participation participation);
+    /** Returns an unmodifiable view of the filtered person list */
+    ObservableList<Person> getFilteredPersonList();
 
     /**
-     * Adds the given participation.
-     * {@code participation} must not already exist in the address book.
+     * Updates the filter of the filtered person list to filter by the given {@code predicate}.
+     * @throws NullPointerException if {@code predicate} is null.
      */
-    public void addParticipation(Participation participation);
+    void updateFilteredPersonList(Predicate<Person> predicate);
+
+
+
+    //// Tutorial-related methods ////
 
     /**
-     * Deletes the given participation.
-     * The participation must exist in the address book.
+     * Returns true if a tutorial with the same identity as {@code tutorial} exists in the address book.
      */
-    void deleteParticipation(Participation target);
+    boolean hasTutorial(Tutorial tutorial);
 
-    /** Returns an unmodifiable view of the filtered person list */
-    ObservableList<Person> getFilteredPersonList();
+    /**
+     * Closes/Deletes the given tutorial.
+     * The tutorial must exist in the address book.
+     */
+    void closeTutorial(Tutorial tutorial);
 
-    /** Returns an unmodifiable view of the filtered tutorial list */
-    ObservableList<Tutorial> getFilteredTutorialList();
+    /**
+     * Retrieves the given tutorial in the list with the same identity.
+     * The tutorial must exist in the address book.
+     */
+    Tutorial getTutorial(Tutorial tutorial);
```

Comment:
Should this method be in the Model level abstraction? Not sure cause it's different from the current methods in model. 
Is this used to retrieve the tutorial in closeTutorialCommand?

Reply:
yep close tutorial parser basically creates a new closetutorialcommand which the constructor takes in a tutorial to close. then when the closetutorialcommand.excute() method is called, it uses that tutorial to retrieve the tutorial with the same subject from the uniquetutoriallist, and then operate on it to remove its list<participation> and subsequently remove the tutorial from the addressbook, so I think that's why we need it to have Model level abstraction also, if not i don't think there is a way to retrieve the list<participation>.


Do let me know if you have better suggestions regarding this.

Reply:
I'm thinking of getTutorialList() into stream filtering, but I'm not sure of the abstraction level at this level because model holds filteredTutorialList

Reply:
ahhh i think that is doable, i will get the entire tutorial list then i will do stream filtering